codejail/safe_exec.py

```diff
@@ -112,14 +113,21 @@ def write(self, *args, **kwargs):
     the_code.append(textwrap.dedent(
         # Execute the sandboxed code.
         """
-        exec code in g_dict
+        exec(code, g_dict)
         """
         # Clean the globals for sending back as JSON over stdout.
         """
         ok_types = (
-            type(None), int, long, float, str, unicode, list, tuple, dict
+            type(None), int, float, str, six.text_type, list, tuple, dict
         )
         bad_keys = ("__builtins__",)
+
+        decoded_dict = {}
+        for k, v in six.iteritems(g_dict):
+            if isinstance(v, bytes):
```

Comment:
Is it possible for the value of a g_dict item to be a dict or other complex type?  eg.  Are there gonna be more byte strings underneath a key that we wouldn't catch with this?

Reply:
`dicts`, `tuples` and `lists` are acceptable types, but we don't recursively go through these structures recursively to make sure that every component is serializable

Reply:
it's a fair point. I can test out a commit to handle that (and probably add a test)